alacritty_terminal/src/ansi.rs

```diff
@@ -1755,4 +1764,45 @@ mod tests {
 
         assert_eq!(handler.color, Some(Rgb { r: 0xf0, g: 0xf0, b: 0xf0 }));
     }
+
+    #[test]
+    fn parse_osc104_reset_color() {
+        let bytes: &[u8] = b"\x1b]104;1;\x1b\\";
+
+        let mut parser = Processor::new();
+        let mut handler = MockHandler::default();
+
+        for byte in bytes {
+            parser.advance(&mut handler, *byte);
+        }
+
+        assert_eq!(handler.reset_color_index, Some(1));
+        assert_eq!(handler.reset_color_count, 1);
+    }
+
+    #[test]
+    fn parse_osc104_reset_all_colors() {
```

Comment:
This test seems awfully complicated for providing such extremely poor coverage of the actual behavior. All you're doing is check that *reset_color` is called 256 times, but for all we know that might all be the same color because during refactoring someone accidentally used `0` instead of an index or something.

In the current state, this test is basically useless other than validating that `reset_color` was called at all. And if that's all you want to do then there's no need to make things so complicated.

Reply:
We could add `Vec` to `MockHandler`  and push all reset colors into that, then check that all expected indices are in the vector.  That would probably also look cleaner than this current iteration.

In a note unrelated to this PR: I think a concept like this could also be applied to the tests I made in #5769 which currently don't cover the case where OSC 4 is called with multiple pairs (e.g. `\e]4;1;#123;2;#456\e\\`)